### PR TITLE
Add new endpoint to get ssh keys from other user

### PR DESCRIPTION
**Pull request by xerbalind**
When updating your user profile it also validates now.
Also updated the ssh_keys regex  to include new key types.

### CODE REVIEW

**Comment by xerbalind** [comment]
workflow fails because `time` crate is outdated, we should update the dependencies.

File: src/controllers/users_controller.rs [hunk 1]
```diff
@@ -58,6 +59,24 @@ pub async fn show_user<'r>(
 	}
 }
 
+#[get("/users/keys/<username>")]
```
[the remark this answers is not in the document]
Comment (xerbalind) [reply src/controllers/users_controller.rs:62]: Sadly leads to routing collisions, but fixable with route ranking.

[the remark this answers is not in the document]
Comment (rien) [reply src/controllers/users_controller.rs:62]: I prefer using ranking to have a cleaner API

**Comment by rien** [comment]
The ssh keys are currently returned in a JSON. Is this on purpose for a specific goal?

I think it would make more sense to just return them in plain text format as you would find them in an `authorized_keys` file (separated by newlines).

We could do some content negotiation and tweak the https://github.com/ZeusWPI/zauth/blob/main/src/views/accepter.rs to also allow for responding to `txt` requests. But that would make it quite complex.

**Review by rien — APPROVED** [review]

**Pull request merged**